### ISSUE TITLE
bap demangling failures.

### ISSUE THREAD

**Issue by dluchaup** [BinaryAnalysisPlatform/bap#312]
I get the following failures when trying to use `bap` with `--demangle`:
```
bap ../tests/strings1.arm-O1.x --demangle=c++filt --use-ida -d asm
Exiting because (Invalid_argument "Expect: arg := (<name> <val>)").
```
Also:
```
bap ../tests/strings1.arm-O1.x --demangle --use-ida -d asm
Exiting because (Invalid_argument "pos + len past end: 34 + 3 > 36").
```

... and:

```
bap ../tests/strings1.arm-O1.x --demangle=arm-linux-gnueabi-c++filt --use-ida -d asm
Exiting because (Invalid_argument "Expect: arg := (<name> <val>)").
```

**Comment by dluchaup** [comment]
FYI, the strings program iscompiled on ubuntu with 
arm-linux-gnueabi-g++ -O1

from

-----
#include <string>
#include <iostream>

using namespace std;
string one() { return string("one"); }
string two() { return string("two"); }

const string g_one_two_init("nothing");
string g_one_two = g_one_two_init;

string one_two() {
  if (g_one_two.compare(g_one_two_init) == 0)
    g_one_two = one();
  return g_one_two += two();
}

int main(int argc, char *argv[]) {
  string str;
  for (int i = 0; i < argc; ++i)
    str +=one_two();
  std::cout << str << std::endl;
  return str.length();
}

**Comment by ivg** [comment]
fixed in #344.

There is no more --demangle option, so nothing to fix :)
